### PR TITLE
Add New Line Lists with Medium Information

### LINKED ISSUE #1517: [FEAT] Specviz: Add air/vacuum to line list metadata and display in plugin
### Description
Currently, `jdaviz/data/linelists/linelist_metadata.json` has information about the source of the list and the units that the wavelengths are in. We should add to this whether the wavelengths are in air or vaccum (since apparently we have some of each) and then display that information in the UI under each list (and probably add it as a column in the `spectral_lines` table as well).

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

@camipacifici and I have been investigating the medium information for the line lists, and it has evolved into a much more difficult task, between trying to track down the original source material, resolving dead links, contacting old colleagues, etc. Some of those (easier ones) were put into #1626. Some additional ones (possibly more controversial) are here. I've officially run out of my timebox on this ticket, so I wanted to close out my initial findings and commit them now. 

This PR makes two changes to our existing line lists:

1. The `Atomic/Ionic` list seems to be a combination two separate lists. We were able to track down the original source material, for the majority of it, which contained the medium info. The remaining portion, however, eludes me. This PR splits the existing Atomic/Ionic list into two separate lists `Atomic/Ionic` and `Atomic/Ionic Fine Structure`, the latter of which we know the medium information for.
2. To address @rosteen's discovery in https://github.com/spacetelescope/jdaviz/pull/1626#pullrequestreview-1102374155, this PR also adds the `Common Galactic` list, which largely contains most of the lines from the Common Stellar (not all of them), but split into two wavelength ranges based on the medium.

To support the above changes, the following adjustments were made to the UI to support these longer list names:

1. ~Apparently the `v-select` we were using for the dropdown has a "whitespace buffer" that `v-combobox` doesn't have. I switched it out to display more of the text name in the dropdown with the same real-estate~ Unfortunately after further testing, the distinction between the two components is that the `v-combobox` is a `v-autocomplete` component; effectively it allows a user to type in anything. With that in mind, a `v-select` is more appropriate here. That being said, a user can still see the list selected by opening the dropdown. 
image
 I think this was a nice-to-have, so it's not a show stopper that we can't get this in. I feel like @rosteen's review is more focused on the point below, but can correct me if I'm wrong
2. I removed our overflow restriction on the actual loaded list `expansion-panel` to show the full list name:

image

Fixes #1517

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by rosteen — COMMENTED** [review]
Thanks for getting a couple more of these in. PR needs to be rebased to `main` to resolve conflicts. Two other comments:

1. It looks like `Atomic/Ionic` isn't available in the preset menu, only `Atomic/Ionic Fine Structute`, I'm not sure why.
2. Now that some of the list names are longer than they used to be, it's probably worth having the text in the preset menu wrap so the wavelengths aren't cut off in the menu at the default plugin width.

**Comment by duytnguyendtn** [comment]
> It looks like Atomic/Ionic isn't available in the preset menu
Yep, that's intended. Referring back to the description, the Atomic/Ionic list only contains the lines I wasn't able to track down. The The A/I Fine Structure one contains the lines I was able to confirm

Good point about the wrapping. I'll look into it

**Comment by rosteen** [comment]
> Yep, that's intended. Referring back to the description, the Atomic/Ionic list only contains the lines I wasn't able to track

Ahh right, got it. Somehow forgot that between reading the description and testing the plugin. Thought it was an air/vac split.

File: jdaviz/data/linelists/linelist_metadata.json
Comment (kecnry) [inline jdaviz/data/linelists/linelist_metadata.json:33]:
```suggestion
    "Citations/Notes": "Split from Atomic/Ionic Fine Structure. JWST Line Lists: The lists and suggested means of implementing them in JWST data analysis software\n A.P.Marston, 30 March 2017\n https://stsci.slack.com/files/anmarston/F4UR94WUX/jwst_line_list.pdf"
```

**Comment by duytnguyendtn** [comment]
@rosteen I've adjusted the UI, let me know if that works for you!

**Comment by duytnguyendtn** [comment]
The docs build failure seems to be unrelated (didn't change the docs); please ignore!

**Comment by kecnry** [comment]
> but is there an easy way to change the preset lists menu to wrap text there as well so it doesn't overflow the bounds of the app

I thought I tested this PR yesterday and didn't see any overflow on my machine with the default plugin width, but I would think we could just enable it with css if necessary (or maybe having ellipsis would be better than wrapping?)

**Comment by duytnguyendtn** [comment]
I think I realized I misunderstood Ricky's earlier message. You can see in my screenshot that there is an overflow of the app's bounds when the menu is dropped down. I wonder if I can anchor the menu to the right, rather than the left...

**Comment by rosteen** [comment]
Sorry, I should have included a screenshot. This is what it looks like for me without messing with anything. Not a huge deal, I don't really think it should block merging this but if it's quick it could be included (sounds like it's not).

Screen Shot 2022-09-22 at 3 36 45 PM

**Comment by duytnguyendtn** [comment]
Haha I think I got it! @rosteen does this work for you?
image

**Review by rosteen — APPROVED** [review]
Looks good to me 👍

File: jdaviz/configs/default/plugins/line_lists/line_lists.vue
Comment (kecnry) [inline jdaviz/configs/default/plugins/line_lists/line_lists.vue:97]: we should probably do this app-wide... how does this work in conjunction with #1673?

Comment (duytnguyendtn) [reply jdaviz/configs/default/plugins/line_lists/line_lists.vue:97]: Good point @kecnry! I've implemented it across all v-selects in https://github.com/spacetelescope/jdaviz/pull/1656/commits/bfa270fb99f5761ef13ff493bcbefdf581302c05

The short answer to your question about how they interact together is that they don't. I WISH they did. Unfortunately #1673 restricts menus to the size of the v-select when you attach. We'll have to think through this case before we merge #1673. I'd advocate for us to get this in as it stands and think about those considerations over in that PR

**Review by kecnry — APPROVED** [review]
LGTM (probably could either use a squash & merge or pre-squashing some of the smaller commits - I'll let you decide)!

**Comment by duytnguyendtn** [comment]
Thanks @kecnry and @rosteen for the eyes! I'll squash and merge (my new default)

**Pull request merged**
